### PR TITLE
[v6r13] Several FTS3 related and other fixes

### PR DESCRIPTION
**Pull request by atsareg**

### CODE REVIEW

File: RequestManagementSystem/Service/ReqManagerHandler.py [hunk 1]
```diff
@@ -113,10 +113,12 @@ def export_putRequest( cls, requestJSON ):
 
       # If there is a waiting Operation with Files
       if op and len( op ):
-        maxWaitingAttempt = max( [ opFile.Attempt for opFile in op if opFile.Status == "Waiting" ] )
-        # In case it is the first attempt, extraDelay is 0
-        # maxWaitingAttempt can be None if the operation has no File, like the ForwardDiset
-        extraDelay = datetime.timedelta( minutes = 2 * math.log( maxWaitingAttempt )  if maxWaitingAttempt else 0 )
+        attemptList = [ opFile.Attempt for opFile in op if opFile.Status == "Waiting" ]
+        if attemptList:
```
Comment (chaen) [inline RequestManagementSystem/Service/ReqManagerHandler.py]: extraDelay is not defined if attemptList is empty. 
And I think in can never be empty since we test len(op)

File: DataManagementSystem/Client/FTSJob.py [hunk 2]
```diff
@@ -729,9 +729,15 @@ def toSQL( self ):
 
     :return: str with SQL fragment
     """
-    colVals = [ ( "`%s`" % column, "'%s'" % value if type( value ) in ( str, datetime.datetime ) else str( value ) )
-                for column, value in self.__data__.items()
-                if value and column not in  ( "FTSJobID", "LastUpdate" ) ]
+    colVals = []
```
Comment (chaen) [inline DataManagementSystem/Client/FTSJob.py]: Why this change ?

File: DataManagementSystem/Client/FTSFile.py [hunk 3]
```diff
@@ -321,10 +321,16 @@ def toJSON( self ):
                       [ val if val != None else "" for val in self.__data__.values() ] ) ) )
   def toSQL( self ):
     """ prepare SQL INSERT or UPDATE statement """
-    colVals = [ ( "`%s`" % column, "'%s'" % value if type( value ) in ( str, datetime.datetime ) else str( value ) )
-                for column, value in self.__data__.items()
-                if value and column not in ( "FTSFileID", "LastUpdate" )  ]
-    colVals.append( ( "LastUpdate", "UTC_TIMESTAMP()" ) )
+    colVals = []
```
Comment (chaen) [inline DataManagementSystem/Client/FTSFile.py]: Why this change ?

File: DataManagementSystem/Client/FTSJob.py (hunk 2 above)
Comment (atsareg) [inline DataManagementSystem/Client/FTSJob.py]: The type of the string must be evaluated as isinstance( value, basestring ). We should all take this on board. In the replaced code the type of value was compared to "str" which is not taking into account Unicode strings. Again, this is what happened in my tests.

File: DataManagementSystem/Client/FTSFile.py (hunk 3 above)
Comment (atsareg) [inline DataManagementSystem/Client/FTSFile.py]: See above

File: DataManagementSystem/Client/FTSJob.py (hunk 2 above)
Comment (chaen) [inline DataManagementSystem/Client/FTSJob.py]: I totally agree with this, I even encourage it, I was just speaking about the refactoring. Readability ?

Comment (atsareg) [inline DataManagementSystem/Client/FTSJob.py]: Did not figure out how to concisely replace this test: type( value ) in ( str, datetime.datetime ) by using basestring which can not be done the same way as str.

Comment (chaen) [inline DataManagementSystem/Client/FTSJob.py]: ok.
I did not have time to do my PR, can include in this one the diff i sent you this morning please ?

Comment (atsareg) [inline DataManagementSystem/Client/FTSJob.py]: Zut, thanks for the reminder ! Have completely forgotten that.
